North-Seattle-College/ad440-winter2022-thursday-repo
# Script to clean All lambda resources

**Overview**
•	This file will allow the user to list and delete the lambda resources through command-line arguments with any manual intervention. The following arguments can be provided at runtime:
o	--region: AWS region where all the lambda resides.
o	--functionName: To specifically delete a certain lambda function in the selected region.
o	–-cleanAllLambda:  To delete all the lambda functions that exist in the selected region.
**Pre-requisites**
•	boto3 installed: pip install boto3
•	AWS CLI properly configured with your credentials: https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html
**To Test**
1.	git checkout Issue#115-Delete-All-Lambda-Resources
2.	cd devops/DeployLambdaAutomation
3.	Run the script with an argument to confirm functionality:
•	To List all the functions: python deleteAllLambdaResources.py --region us-west-2
•	To delete only specific function : python deleteAllLambdaResources.py --region us-west-2 --functionName [specific table Name]
**WARNING: The below command will delete all the lambda functions. Make sure before running the command that you are not deleting other people’s functions that are in use.**
•	To delete all the functions in the region: python deleteAllLambdaResources.py --region us-west-2 –cleanAllLambda

5.	You should see the below in the terminal
 
![image](https://user-images.githubusercontent.com/72327494/158699530-d0edb6fd-17aa-4cc3-9ad2-b681d48b8633.png)
![image](https://user-images.githubusercontent.com/72327494/158699554-29b317c2-794c-4e09-b386-85065a8ea026.png)
![image](https://user-images.githubusercontent.com/72327494/158699577-96e0aa20-9fce-4c56-a387-bd58279aea29.png)

 6. To confirm all the deletion login to AWS console
 

<html>
<body>
<h3 dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.25em; font-weight: 600; line-height: 1.25; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Time Spent</h3>

Activity | Estimate | Actual Time Spent
-- | -- | --
Research  | 2 hours | 3 hours
Automation | 3 hours | 5 hours
Testing | 1 hour | 1.30 hours
Documentation and PR | 1hour | 1 hours


</body>
</html>
